AirCasting/SessionViews/SessionHeaderView.swift

```diff
@@ -48,18 +48,54 @@ struct SessionHeaderView: View {
 
 private extension SessionHeaderView {
     var dateAndTime: some View {
-        guard let start = session.startTime else {
-            return Text("")
-        }
-        let end = session.endTime ?? Date()
         
         let formatter = DateIntervalFormatter()
-        
         formatter.timeStyle = .short
         formatter.dateStyle = .medium
-        let string = DateIntervalFormatter().string(from: start, to: end)
-        let replaced = string.replacingOccurrences(of: "—", with: "-")
-        return Text(replaced)
+        var date = ""
+        var additionalDate = ""
+        
+            guard let start = session.startTime else { return Text("") }
+            let end = session.endTime ?? Date()
+            let string = DateIntervalFormatter().string(from: start, to: end)
+        
+        if is24Hour() {
+            let replaced = string.replacingOccurrences(of: "—", with: "-")
+            return Text(replaced)
+        } else {
+            if string.contains("–") {
+                let time = string.components(separatedBy: "–")
+                var timeLast = time.last
+                // some of the dates on fixed session are different we need to handle that 'out of nowhere (few days sessions)' case
+                if ((time.last?.contains("/")) != nil) {
+                    additionalDate = (time.last?.components(separatedBy: ",").first)!
+                    timeLast = time.last?.components(separatedBy: ",").last
+                }
+                
+                let endTime12 = timeLast!.trimmingCharacters(in: .whitespaces)
+            
+                let last = time.first!
+                let time2 = last.components(separatedBy: ",")
+                date = time2.first!
+                var startTime12 = time2.last?.trimmingCharacters(in: .whitespaces)
+                
+                if !(startTime12!.contains("PM") || startTime12!.contains("AM")) {
+                    (endTime12.contains("PM")) ? startTime12?.append(" PM") : startTime12?.append(" AM")
+                }
+                if additionalDate == "" {
+                    date.append(", \(timeConversion24(time12: startTime12!))-\(timeConversion24(time12: endTime12))")
+                } else {
+                    date.append(", \(timeConversion24(time12: startTime12!))-\(additionalDate), \(timeConversion24(time12: endTime12))")
+                }
```

Comment:
i feel like this whole block of code can be handled by a well formatted `DateFormatter`. Not sure tho as it's pretty enigmatic in what it does. If it needs to stay like this - please comment it heavily with examples so that anyone that stumbles upon it will have an easier time understanding what it does.

Example:
```Swift
// Date format provided can be either "2021/09/09 10:20PM" or "2021-09-09 22:20" so we
// need to distinguish between those two formats (etc. etc)...
```